File: server_nodejs/src/api/cart/cart.controller.js
const service = require('./cart.service');

module.exports = {
    getCart: async (req, res, next) => {
        try {
            const user = req.user;
            const DTO = await service.getCart(user.uid);
            res.status(DTO.statusCode).json(DTO);
        } catch (err) {
            next(err);
        }
    },
    updateCart: async (req, res, next) => {
        try {
            const user = req.user;
            const DTO = await service.updateCart(
                user.uid,
                req.body.item,
                req.body.quantity,
            );
            res.status(DTO.statusCode).json(DTO);
        } catch (err) {
            next(err);
        }
    },
    deleteCart: async (req, res, next) => {
        try {
            const user = req.user;
            const DTO = await service.deleteCart(user.uid);
            res.status(DTO.statusCode).json(DTO);
        } catch (err) {
            next(err);
        }
    },
    createCartItem: async (req, res, next) => {
        try {
            const user = req.user;
            const DTO = await service.createCartItem(user.uid, req.body.items);
            res.status(DTO.statusCode).json(DTO);
        } catch (err) {
            next(err);
        }
    },
    getDiscount: async (req, res, next) => {
        try {
            const user = req.user;
            const DTO = await service.getDiscount(user.uid, req.query.code);
            res.status(DTO.statusCode).json(DTO);
        } catch (error) {
            next(error);
        }
    },
    checkout: async (req, res, next) => {
        try {
            const user = req.user;
            const DTO = await service.checkout(user.uid, req.body);
            res.status(DTO.statusCode).json(DTO);
        } catch (error) {
            next(error);
        }
    },
};

File: server_asp/src/api/cart/CartController.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;
using shieldtify.common;
using shieldtify.middleware;
using shieldtify.models;

namespace shieldtify.api.cart
{
    public static class CartController
    {
        [Tags("Cart")]
        public static APIRes getCart(HttpContext context)
        {
            try
            {
                return Middleware.MiddlewareAuthorize(() =>
                {
                    var user = (ClientAccount?)context.Items["User"];
                    var DTO = CartService.getCart(user.Uid.ToString());
                    context.Response.StatusCode = DTO.statusCode;
                    return DTO;
                }, context, new List<string> { "client" });
            }
            catch (Exception)
            {
                throw;
            }
        }
        [Tags("Cart")]
        public static APIRes updateCart(HttpContext context, [FromBody] UpdateCartBody body)
        {
            try
            {
                return Middleware.MiddlewareAuthorize(() =>
                {
                    var user = (ClientAccount?)context.Items["User"];
                    var DTO = CartService.updateCart(user.Uid.ToString(), body.item, body.quantity);
                    context.Response.StatusCode = DTO.statusCode;
                    return DTO;
                }, context, new List<string> { "client" });
            }
            catch (Exception)
            {
                throw;
            }
        }
        [Tags("Cart")]
        public static APIRes deleteCart(HttpContext context)
        {
            try
            {
                return Middleware.MiddlewareAuthorize(() =>
                {
                    var user = (ClientAccount?)context.Items["User"];
                    var DTO = CartService.deleteCart(user.Uid.ToString());
                    context.Response.StatusCode = DTO.statusCode;
                    return DTO;
                }, context, new List<string> { "client" });
            }
            catch (Exception)
            {
                throw;
            }
        }

        [Tags("Cart")]
        public static APIRes createCartItem(HttpContext context, [FromBody] AddCartItemBody body)
        {
            try
            {
                return Middleware.MiddlewareAuthorize(() =>
                {
                    var user = (ClientAccount?)context.Items["User"];
                    var DTO = CartService.createCartItem(user.Uid.ToString(), body.items);
                    context.Response.StatusCode = DTO.statusCode;
                    return DTO;
                }, context, new List<string> { "client" });
            }
            catch (Exception)
            {
                throw;
            }
        }
        [Tags("Cart")]
        public static APIRes getDiscount(HttpContext context, [FromQuery] string code)
        {
            try
            {
                return Middleware.MiddlewareAuthorize(() =>
                {
                    var user = (ClientAccount?)context.Items["User"];
                    var DTO = CartService.getDiscount(user.Uid.ToString(), code);
                    context.Response.StatusCode = DTO.statusCode;
                    return DTO;
                }, context, new List<string> { "client" });
            }
            catch (Exception)
            {
                throw;
            }
        }
        [Tags("Cart")]
        public static APIRes checkout(HttpContext context, [FromBody] CheckoutBody body)
        {
            try
            {
                return Middleware.MiddlewareAuthorize(() =>
                {
                    var user = (ClientAccount?)context.Items["User"];
                    var DTO = CartService.checkout(user.Uid.ToString(), body);
                    context.Response.StatusCode = DTO.statusCode;
                    return DTO;
                }, context, new List<string> { "client" });
            }
            catch (Exception)
            {
                throw;
            }
        }
    }

    public class UpdateCartBody
    {
        public required string item { get; set; }
        public required int quantity { get; set; }
    }

    public class AddCartItemBody
    {
        public required List<Items> items { get; set; }
    }

    public class Items
    {
        public required string item { get; set; }
        public required int quantity { get; set; }
    }

    public class CheckoutBody
    {
        public string? code { get; set; }
        public required string payment_method { get; set; }
        public required string receive_method { get; set; }
        public required string shipping_addressid { get; set; }
    }
}